frontend/src/registration/components/CustomDomainSetupPage/CustomDomainSetupPage.tsx

```diff
@@ -0,0 +1,139 @@
+import { RegistrationSteps } from 'global/constants';
+import { RootState } from 'global/state';
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { WrappedMessage } from 'utils/intl';
+import { RedirectToCorrectStep } from 'registration/components';
+import { RegistrationStateModel } from 'registration/models';
+import { Row, Col, Button, Spinner } from 'react-bootstrap';
+import { goToNextStep } from '../../actions';
+import { RegistrationPage } from '../RegistrationPage';
+import messages from './displayMessages';
+import './styles.scss';
+
+interface ActionProps {
+  goToNextStep: Function;
+}
+
+interface State {
+  externalDomain: string;
+}
+
+interface Props extends ActionProps {}
+interface StateProps extends RegistrationStateModel {}
+
+interface Props extends StateProps, ActionProps {}
+
+@connect<{}, ActionProps, {}, Props, RootState>(
+  (state: RootState) => ({
+    ...state.registration
+  }),
+  {
+    goToNextStep
+  }
+)
+export class CustomDomainSetupPage extends React.PureComponent<Props, State> {
+  constructor(props: Props, state: State) {
+    super(props);
+  }
+
+  public render() {
+    const stepBox = (step: number, instruction: any, extra?: any) => {
+      return (
+        <div className="instruction-box">
+          <Row>
+            <Col lg="1" className="number">
+              {step}
+            </Col>
+            <Col lg="11" className="instruction">
+              {instruction}
+            </Col>
+          </Row>
+          {extra && (
+            <Row>
+              <Col className="extra">{extra}</Col>
+            </Row>
+          )}
+        </div>
+      );
+    };
```

Comment:
This could just be implemented as another component that in internal to this file. It would clarify the code a little bit.

Reply:
Done!